tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteConsumerGroupsTest.java

```diff
@@ -17,279 +17,448 @@
+            this.syncCommit = syncCommit;
+        }
+
+        void configure() {
+            configured = true;
+            configure(props);
+            customPropsOpt.ifPresent(props::putAll);
+            consumer = new KafkaConsumer<>(props);
+        }
+
+        void configure(Properties props) {
+            props.put("bootstrap.servers", broker);
+            props.put("group.id", groupId);
+            props.put("key.deserializer", StringDeserializer.class.getName());
+            props.put("value.deserializer", StringDeserializer.class.getName());
+        }
+
+        abstract void subscribe();
+
+        @Override
+        public void run() {
+            assert configured : "Must call configure before use";
+            try {
+                subscribe();
+                while (!isShutdown) {
+                    consumer.poll(Duration.ofMillis(Long.MAX_VALUE));
+                    if (syncCommit)
+                        consumer.commitSync();
+                }
+            } catch (WakeupException e) {
+                // OK
+            } finally {
+                consumer.close();
+            }
+        }
+
+        void shutdown() {
+            isShutdown = true;
+            consumer.wakeup();
+        }
+    }
 
-        ConsumerGroupCommand.ConsumerGroupService service2 = getConsumerGroupService(cgcArgs);
-        Map<String, Throwable> result = service2.deleteGroups();
-        assertTrue(result.size() == 2 &&
-                result.containsKey(GROUP) && result.get(GROUP) == null &&
-                result.containsKey(missingGroup) &&
-                result.get(missingGroup).getMessage().contains(Errors.GROUP_ID_NOT_FOUND.message()),
-            "The consumer group deletion did not work as expected");
+    static class ConsumerRunnable extends AbstractConsumerRunnable {
+        final String topic;
+        final String groupProtocol;
+        final String strategy;
+        final Optional<String> remoteAssignor;
+
+        public ConsumerRunnable(String broker,
+                                String groupId,
+                                String groupProtocol,
+                                String topic,
+                                String strategy,
+                                Optional<String> remoteAssignor,
+                                Optional<Properties> customPropsOpt,
+                                boolean syncCommit) {
+            super(broker, groupId, customPropsOpt, syncCommit);
+
+            this.topic = topic;
+            this.groupProtocol = groupProtocol;
+            this.strategy = strategy;
+            this.remoteAssignor = remoteAssignor;
+        }
+
+        @Override
+        void configure(Properties props) {
+            super.configure(props);
+            props.put(ConsumerConfig.GROUP_PROTOCOL_CONFIG, groupProtocol);
+            if (groupProtocol.toUpperCase(Locale.ROOT).equals(GroupProtocol.CONSUMER.toString())) {
```

Comment:
This could be simplified to `if (groupProtocol == GroupProtocol.CONSUMER)` if we work with the `GroupProtocol` object instead of a string version. We truly only need the string when passing it into the consumer prop `GROUP_PROTOCOL_CONFIG`  above.